### ISSUE TITLE
Question: Is it possible to obtain the last n minutes of a TimeSeries?

### ISSUE THREAD

**Issue by patryk0605** [esnet/pond#56]
Assuming I have a TimeSeries already created with the following range for example: 

1. Mon Nov 14 2016 06:30:00 GMT-0500 (Eastern Standard Time)
2. Mon Nov 14 2016 08:00:00 GMT-0500 (Eastern Standard Time)

Is it possible to get a range of the last 15 minutes using pond? So the new range would be:

1. Mon Nov 14 2016 07:45:00 GMT-0500 (Eastern Standard Time)
2. Mon Nov 14 2016 08:00:00 GMT-0500 (Eastern Standard Time)

**Comment by patryk0605** [comment]
This is the current way I am doing it now: 

```js
new TimeRange(moment(trafficSeries.end()).subtract(15, 'minutes'), moment(trafficSeries.end()))
```

Was just wondering if there is a built in method.

**Comment by pjm17971** [comment]
No, there's not. But that seems reasonable to add. I'm assuming what you're interesting in is more like `timeseries.last("15m")` -> new `TimeSeries` than TimeRange.last("15m") -> a `TimeRange`?

**Comment by patryk0605** [comment]
@pjm17971 Yes, something along the lines of ```timeseries.last("15m")``` would work very well! Not a priority, but nice to have feature.

**Comment by cathal-killeen** [comment]
Has anyone been working on this @pjm17971? I would be interested in such a feature and would love to implement it. Ideally, I think the method API should follow the various [Moment duration](https://momentjs.com/docs/#/durations/) formats. Arbitrary durations could then easily be requested, such as `timeseries.last({ minutes: 15, seconds: 30 })`

**Comment by pjm17971** [comment]
Thanks for the offer to work on this. This one is a little tricky, because of the typescript port. In particular, instead of periods (or durations) being specified as just a string, e.g. "1h" they will be a period("1d"). That constructor actually then work with either a ms duration, a string like above, a moment.duration or a moment.duration style object (as you described).

```
        const m = moment.duration(24, "hours");
        const p = period(m);
        expect(+p).toBe(86400000);

      // OR

        const p = period({
            seconds: 2,
            minutes: 2,
            hours: 2,
            days: 2,
            weeks: 2,
            months: 2,
            years: 2
        });
        expect(+p).toBe(69732122000);
```

That said, I haven't implemented this specific item. I'll ping you here when there's a branch here that you could implement that on top of without it being difficult to port over to typescript in the near future. There's a collection implementation in Typescript, but no TimeSeries on top of that yet.

It's pretty simple, because you can substract +period from the end time of the timeseries and use that time with `bisect` or `atTime` to get the index, and then `slice` to the end.
